File: bmpm-php/BMPMAshkenazi.php
<?php

class BMPMAshkenazi extends BMPM {
	function __construct() {
		$this->setType(BMPM::TYPE_ASHKENAZI);
		
		include "ash/approxcommon.php";
		$this->setApproxCommon($approxCommon);
		
		include "ash/exactcommon.php";
		$this->setExactCommon($exactCommon);
		
		include "ash/hebrewcommon.php";
		$this->setHebrewCommon($hebrewCommon);

		include "ash/languagenames.php";
		$this->setLanguageNames($languages);
		
		include "ash/lang.php";
		$this->setAllLanguagesBitmap($all);
		$this->setLanguageRules($languageRules);

		$rules = array();
		include "ash/rulesany.php";
		include "ash/rulescyrillic.php";
		include "ash/rulesenglish.php";
		include "ash/rulesfrench.php";
		include "ash/rulesgerman.php";
		include "ash/ruleshebrew.php";
		include "ash/ruleshungarian.php";
		include "ash/rulespolish.php";
		include "ash/rulesromanian.php";
		include "ash/rulesrussian.php";
		include "ash/rulesspanish.php";
		$this->setRules($rules);

		$exact = array();
		include "ash/exactany.php";
		include "ash/exactcyrillic.php";
		include "ash/exactenglish.php";
		include "ash/exactfrench.php";
		include "ash/exactgerman.php";
		include "ash/exacthebrew.php";
		include "ash/exacthungarian.php";
		include "ash/exactpolish.php";
		include "ash/exactromanian.php";
		include "ash/exactrussian.php";
		include "ash/exactspanish.php";
		$this->setExact($exact);

		$approx = array();
		include "ash/approxany.php";
		include "ash/approxcyrillic.php";
		include "ash/approxenglish.php";
		include "ash/approxfrench.php";
		include "ash/approxgerman.php";
		include "ash/approxhebrew.php";
		include "ash/approxhungarian.php";
		include "ash/approxpolish.php";
		include "ash/approxromanian.php";
		include "ash/approxrussian.php";
		include "ash/approxspanish.php";
		$this->setApprox($approx);

		return $this;
	}
}

?>

File: bmpm-cpp/BMPMAshkenazi.cpp
#include "cpp/php2cpp.h"


#ifndef BMPMASHKENAZI_CPP
#define BMPMASHKENAZI_CPP

#include "BMPMAshkenazi.h"

BMPMAshkenazi::BMPMAshkenazi()
    {
        this->setType(BMPM::TYPE_ASHKENAZI);
        #include "ash/approxcommon.h"
        this->setApproxCommon(approxCommon);
        #include "ash/exactcommon.h"
        this->setExactCommon(exactCommon);
        #include "ash/languagenames.h"
        this->setLanguageNames(languages);
        #include "ash/lang.h"
        this->setAllLanguagesBitmap(all);
        this->setLanguageRules(languageRules);
        rules = {};
        #include "ash/rulesany.h"
        #include "ash/rulescyrillic.h"
        #include "ash/rulesenglish.h"
        #include "ash/rulesfrench.h"
        #include "ash/rulesgerman.h"
        #include "ash/ruleshebrew.h"
        #include "ash/ruleshungarian.h"
        #include "ash/rulespolish.h"
        #include "ash/rulesromanian.h"
        #include "ash/rulesrussian.h"
        #include "ash/rulesspanish.h"
        this->setRules(rules);
        exact = {};
        #include "ash/exactany.h"
        #include "ash/exactcyrillic.h"
        #include "ash/exactenglish.h"
        #include "ash/exactfrench.h"
        #include "ash/exactgerman.h"
        #include "ash/exacthebrew.h"
        #include "ash/exacthungarian.h"
        #include "ash/exactpolish.h"
        #include "ash/exactromanian.h"
        #include "ash/exactrussian.h"
        #include "ash/exactspanish.h"
        this->setExact(exact);
        approx = {};
        #include "ash/approxany.h"
        #include "ash/approxcyrillic.h"
        #include "ash/approxenglish.h"
        #include "ash/approxfrench.h"
        #include "ash/approxgerman.h"
        #include "ash/approxhebrew.h"
        #include "ash/approxhungarian.h"
        #include "ash/approxpolish.h"
        #include "ash/approxromanian.h"
        #include "ash/approxrussian.h"
        #include "ash/approxspanish.h"
        this->setApprox(approx);
        return this;
    }

#else
#endif